RubrikPolaris/M365/Start-OperationalRecovery.ps1

```diff
@@ -0,0 +1,211 @@
+function Start-OperationalRecovery() {
+    <#
+    .SYNOPSIS
+    Operational restore Exchange for an AD Group 
+    
+    .DESCRIPTION
+    Operational restore Exchange for an AD Group from latest backups before a
+    given recovery point in time using the mailboxTimeRange.
+
+    .PARAMETER Name
+    The name of the operational recovery you wish to choose. 
+
+    .PARAMETER RecoveryPoint
+    The date time you wish to use to restore closest earlier backups. The format
+    is "YYYY-MM-DD HH:MM:SS"
+    
+    .PARAMETER SubscriptionName
+    The subscription name you wish to operational restore under.
+   
+    .PARAMETER AdGroupId
+    The ID of the AD Group you wish to operational restore.
+
+    .PARAMETER WorkloadType
+    The type of workload you wish to mass restore, only "Exchange" is supported
+    right now.
+  
+    .PARAMETER MailboxFromTime
+    The date time you wish to use to retore the emails received after that. The
+    format is "YYYY-MM-DD HH:MM:SS"
+
+    .PARAMETER MailboxUntilTime
+    The date time you wish to use to retore the emails received before that. The
+    format is "YYYY-MM-DD HH:MM:SS"
+
+    .PARAMETER ArchiveFolderAction
+    The Action of archive folder you wish to use to restore mailbox.
+
+    .PARAMETER SubWorkloadType
+    The type of sub workload you wish to restore. Only supported for "Exchange"
+    workload type, where sub workload types are "Calendar", "Contacts" and "Mailbox"
+    
+    .INPUTS
+    None. You cannot pipe objects to Start-OperationalRecovery.
+   
+    .OUTPUTS
+    System.Object. The taskchainID, massRecoveryInstanceID, error and jobID for 
+    the mass recovery job.
+   
+    .EXAMPLE
+    PS> Start-OperationalRecovery -Name $name -RecoveryPoint $recoveryPoint -SubscriptionName $subscriptionName -AdGroupId $adGroupId -WorkloadType $workloadType -MailboxFromTime $mailboxFromTime -MailboxUntilTime $mailboxUntilTime
+    
+    PS> Start-OperationalRecovery -Name $name -RecoveryPoint $recoveryPoint -SubscriptionName $subscriptionName -AdGroupId $adGroupId -WorkloadType $workloadType -ArchiveFolderAction $archiveFolderAction 
+   
+    #>
+
+    param(
+        [Parameter(Mandatory=$True)]
+        [String]$Name,
+        [Parameter(Mandatory=$True)]
+        [DateTime]$RecoveryPoint,
+        [Parameter(Mandatory=$True)]
+        [String]$SubscriptionName,
+        [Parameter(Mandatory=$True)]
+        [String]$AdGroupId,
+        [Parameter(Mandatory=$True)]
+        [ValidateSet("Exchange")]
+        [String]$WorkloadType,
+        [Parameter(Mandatory=$False)]
+        [DateTime]$MailboxFromTime,
+        [Parameter(Mandatory=$False)]
+        [DateTime]$MailboxUntilTime,
+        [Parameter(Mandatory=$False)]
+        [ValidateSet("NO_ACTION", "EXCLUDE_ARCHIVE", "ARCHIVE_ONLY")]
+        [String]$ArchiveFolderAction,
+        [Parameter(Mandatory=$False)]
+        [ValidateSet("Mailbox", "Calendar", "Contacts")]
+        [String]$SubWorkloadType,
```

Comment:
add `[Parameter(Mandatory=$True)]` for `SubWorkloadType`

Reply:
@rish-singhal I think we should not add [Parameter(Mandatory=$True)] for SubWorkloadType, if given empty, means the user want to run operational recovery for the all the subTypes of exchange, or specify just one subWorkType(though this is not supported in UI workflow now)

Reply:
but `ArchiveFolderAction` and `MailboxFromTime` are only applicable for Mailbox case right? how will we incorporate filters for Calendar, Contacts?

Reply:
Contacts does not have filters, and Calendars we only not provide filters to customers, currently are set default to two weeks from now. So if SubworkloadType is set to Contacts or Calendars, because we have this check  ```
$snappableToSubSnappableMap[$WorkloadType] | Where-Object {
        ($_.NameSuffix -eq $SubWorkloadType) -or ($SubWorkloadType -eq "")
```
we wont run ArchiveFolderAction and MailboxFromTime these filters for mailbox, so it is okay.